src/pixel/contiguous_pixel.rs

```diff
@@ -0,0 +1,45 @@
+use crate::*;
+
+/// An trait for converting to pixel from a slice.
+pub trait ContiguousPixel: HomogeneousPixel {
+    /// Returns a reference to the pixel type from a slice with the same number of components.
+    fn from_component_slice_ref(slice: &[Self::Component]) -> &Self;
```

Comment:
or const generic for COMPONENT_COUNT, and leave slice to array conversion to the caller?

Reply:
Hmmm, isn't this the inverse of `AsSlice`? Maybe conversion in both directions could be more similar?

Reply:
This trait was only added since the `ImageBuffer` in `image` stores it as `Vec<P::Subpixel>` instead of `Vec<P>` for some reason and so a conversion is needed for the `get_pixel(x, y)` functions to work.

Switching to `TryFrom` sounds like a good idea as it saves us from creating this trait in the first place though it might look less readable in function where clauses but I don't feel too strongly about that.

> or const generic for COMPONENT_COUNT, and leave slice to array conversion to the caller?

That saves us even more work as the user can already call `<P as HomogeneousPixel>::from_components(array), but I don't see any reason we can't also implement `TryFrom<&[T]>` as well which calls this inside.